### PR TITLE
feat: Support models without fields.

### PR DESCRIPTION
**Pull request by SandPod**
It adds support for models without any defined fields.

These models could be used for streaming connections to signal different events.

Fixes this already closed issue: #2454

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [x] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

None - this is a new feature.

### CODE REVIEW

File: tools/serverpod_cli/lib/src/analyzer/models/model_parser/model_parser.dart [hunk 1]
```diff
@@ -152,10 +141,21 @@ class ModelParser {
...
+    if (fieldsNode is! YamlMap) return fields;
+
+    fields.addAll(fieldsNode.nodes.entries.expand((fieldNode) {
+      return _parseModelFieldDefinition(
+        fieldNode,
+        docsExtractor,
+        extraClasses,
+        serverOnlyClass,
+      );
+    }).toList());
```
Comment (hampuslavin) [inline tools/serverpod_cli/lib/src/analyzer/models/model_parser/model_parser.dart:157]: Style question: Do you generally prefer the full function syntax in this project over the shorthand arrow?

File: tests/serverpod_test_server/test_integration/database_operations/crud/create_test.dart [hunk 2]
```diff
@@ -82,4 +82,15 @@ void main() async {
     expect(inserted.id, isNotNull);
     expect(inserted.num, 1);
   });
+
+  test('Given a model without fields when inserting then the model is created.',
```
Comment (hampuslavin) [inline tests/serverpod_test_server/test_integration/database_operations/crud/create_test.dart:86]: nit 
```suggestion
  test('Given a model without fields when inserting it then the model is created.',
```

**Review by hampuslavin — APPROVED** [review]
🚀 🌟

File: tools/serverpod_cli/lib/src/analyzer/models/model_parser/model_parser.dart (hunk 1 above)
Comment (SandPod) [reply tools/serverpod_cli/lib/src/analyzer/models/model_parser/model_parser.dart:157]: Not really, but we use function syntax if the body is multiple lines long.
This usually helps with code readability.

**Pull request merged**